File: test-suite/src/test/java/io/micronaut/kafka/docs/consumer/errors/ExceptionProductListener.java
package io.micronaut.kafka.docs.consumer.errors;

import io.micronaut.configuration.kafka.annotation.ErrorStrategy;
import io.micronaut.configuration.kafka.annotation.ErrorStrategyValue;
import io.micronaut.configuration.kafka.annotation.KafkaListener;
import io.micronaut.context.annotation.Requires;

@Requires(property = "spec.name", value = "ExceptionProductListenerTest")
// tag::annotation[]
@KafkaListener(
        value = "myGroup",
        errorStrategy = @ErrorStrategy(
                value = ErrorStrategyValue.RETRY_ON_ERROR,
                retryDelay = "50ms",
                retryCount = 3,
                exceptionTypes = { MyException.class, MySecondException.class }
        )
)
// end::annotation[]
public class ExceptionProductListener {
}

class MyException extends Exception {}
class MySecondException extends Exception {}

File: test-suite-python/src/test/python/micronaut/kafka/docs/consumer/errors/ExceptionProductListener.py
from micronaut.configuration.kafka.annotation import ErrorStrategy, ErrorStrategyValue, KafkaListener
from micronaut.context.annotation import Requires


class MyException(Exception):
    pass


class MySecondException(Exception):
    pass


@Requires(property="spec.name", value="ExceptionProductListenerTest")
# tag::annotation[]
@KafkaListener(
    value="myGroup",
    errorStrategy=ErrorStrategy(
        value=ErrorStrategyValue.RETRY_ON_ERROR,
        retryDelay="50ms",
        retryCount=3,
        exceptionTypes=[MyException, MySecondException]
    )
)
# end::annotation[]
class ExceptionProductListener:
    pass
